tools/test-generator/dune-project

```diff
@@ -1 +1 @@
-(lang dune 1.0)
+(lang dune 1.6)
```

Comment:
So dune 1.6 is required for the test generator, but for exercises it's generally 1.1 (with the exception of palindrome-products that has 1.0). Is this because of features in 1.6 that are necessary to build your additions to the test generator, or a why-not version bump? I suppose that keeping the exercises at 1.1 is fine until we have a particular need.